unidist/core/backends/mpi/core/communication.py

```diff
@@ -53,6 +54,31 @@ def get_mpi_state():
     return comm, rank, world_size
 
 
+def get_topology():
+    """
+    Get topology of MPI cluster.
+
+    Returns
+    -------
+    dict
+        Dictionary, containing workers ranks assignments by IP-addresses in
+        the form: {hode_ip0: [rank_2, rank_3, ..], node_ip1: [rank_i, ..], ..}.
+    """
+    import socket
+
+    topology = defaultdict(list)
+    comm, rank, _ = get_mpi_state()
+    hostname = socket.gethostname()
+    host = socket.gethostbyname(hostname)
+    cluster_info = comm.allgather((host, rank))
```

Comment:
What information does cluster_info contain here? In other words, what is the return value of comm.allgather((host, rank)) here?

Reply:
`cluster_info` contains pairs of (host_ip, rank) for all ranks (example for 4 workers+executor+monitor):
```python
[('10.125.130.30', 0), ('10.125.130.30', 1), ('10.125.130.30', 2), ('10.125.130.30', 3), ('10.125.130.32', 4), ('10.125.130.32', 5)]
```

Reply:
ok